packages/discovery-react-components/src/utils/document/transformEnrichment.ts

```diff
@@ -146,14 +147,11 @@ function getAllAttributesInRelation({ relations }: Relations): Attributes[] {
 // Function checks if attribute is valid and has location data in it.
 // Since have observed noisy data before so this function is required.
 function isValidAttribute(attr: Attributes): boolean {
-  if (attr && attr.location && attr.location.begin && attr.location.end) {
-    return true;
-  }
-  return false;
+  return !!(attr && attr.location && attr.location.begin && attr.location.end);
 }
 
 export default function transformEnrichment(enrichedHtml: EnrichedHtml[]): EnrichedHtml[] {
-  const newEnrichedHtml = { ...enrichedHtml };
+  const newEnrichedHtml = cloneDeep(enrichedHtml);
```

Comment:
Why do we need to clone?

Reply:
In order to maintain immutability created a deep cloned copy and returning it after adding metadata/attributes/relations on top of it.

Reply:
But we're already doing a `cloneDeep` in `processDoc`, before calling this function. So this is probably not necessary.